### PR TITLE
[DNR] update declarativenetrequest documentation with global rules

### LINKED ISSUE #216: [DNR] Update declarativeNetRequest documentation page
Update the declarativeNetRequest page so it includes the documentation for the global rule limit that were updated in chromium but were not ported over to developer.chrome.com

### PR DESCRIPTION
**Pull request by Celsius273**
Fixes #216

This PR updates the declarativeNetRequest page with changes from crrev.com/c/2563238 in chrome/common/extensions/docs/templates/intros/declarativeNetRequest.html that did not make it to the new developer.chrome.com site.

### CODE REVIEW

**Comment by Celsius273** [comment]
@googlebot I signed it!

**Comment by Celsius273** [comment]
@googlebot I consent.

**Comment by robdodson** [comment]
@awfuchs are you cool reviewing this?

File: site/en/docs/extensions/reference/declarativeNetRequest/index.md [hunk 1]
```diff
@@ -40,12 +40,22 @@ of type [Ruleset][4], as shown below.
 ## Rule Resources
 
 An extension can specify up to [MAX_NUMBER_OF_STATIC_RULESETS][5] [rulesets][6] as part of the
-`"rule_resources"` manifest key. The number of rules across **enabled** static rulesets must be less
-than the [MAX_NUMBER_OF_RULES][7] constant.
+`"rule_resources"` manifest key. An extension is allowed to enable up to at least
```
Comment (awfuchs) [inline site/en/docs/extensions/reference/declarativeNetRequest/index.md:43]: What does "up to at least" mean? I read "up to" as an upper limit, and "at least" as a lower limit...

File: site/en/docs/extensions/reference/declarativeNetRequest/index.md [hunk 2]
```diff
@@ -40,12 +40,22 @@ of type [Ruleset][4], as shown below.
...
+[GUARANTEED_MINIMUM_STATIC_RULES][7] static rules. Additional static rule sets may or may not be
+enabled depending on the available [global static rule limit][8].
+
+## Global Static Rule Limit
+
+In addition to the [GUARANTEED_MINIMUM_STATIC_RULES][9] static rules guaranteed for each extension,
+extensions can enable additional static rulesets depending on the available global static rule limit. This
+global limit is shared between all extensions and can be used by extensions on a first come first
```
Comment (awfuchs) [inline site/en/docs/extensions/reference/declarativeNetRequest/index.md:51]: "first-come, first-served"

File: site/en/docs/extensions/reference/declarativeNetRequest/index.md [hunk 3]
```diff
@@ -40,12 +40,22 @@ of type [Ruleset][4], as shown below.
...
+serve basis. Extensions shouldn't depend on the global limit having a specific value and should
+instead rely the [getAvailableStaticRuleCount][10] API method to find the additional rule limit available to
```
Comment (awfuchs) [inline site/en/docs/extensions/reference/declarativeNetRequest/index.md:53]: rely -> use

File: site/en/docs/extensions/reference/declarativeNetRequest/index.md [hunk 4]
```diff
@@ -40,12 +40,22 @@ of type [Ruleset][4], as shown below.
...
+them.
 
 ## Rules
 
-A single declarative [Rule][8] consists of four fields: `id`, `priority`, `condition` and `action`.
+A single declarative [Rule][11] consists of four fields: `id`, `priority`, `condition` and `action`.
```
Comment (awfuchs) [inline site/en/docs/extensions/reference/declarativeNetRequest/index.md:58]: `condition`, and
https://developers.google.com/style/commas#serial-commas

File: site/en/docs/extensions/reference/declarativeNetRequest/index.md [hunk 5]
```diff
@@ -40,12 +40,22 @@ of type [Ruleset][4], as shown below.
...
 There are the following kinds of rules:
```
Comment (awfuchs) [inline site/en/docs/extensions/reference/declarativeNetRequest/index.md:59]: It looks very odd to say there is a following list of one thing...

File: site/en/docs/extensions/reference/declarativeNetRequest/index.md [hunk 6]
```diff
@@ -78,18 +88,18 @@ the request URL. Some examples of URL filters:
 
 ## Dynamic rules
 
-To add or remove rules dynamically, extensions can use the [updateDynamicRules][9] API method.
+To add or remove rules dynamically, extensions can use the [updateDynamicRules][12] API method.
```
Comment (awfuchs) [inline site/en/docs/extensions/reference/declarativeNetRequest/index.md:91]: s/extensions can//

To add or remove rules dynamically, use the...

**Comment by Celsius273** [comment]
@googlebot I consent.

File: site/en/docs/extensions/reference/declarativeNetRequest/index.md (hunk 1 above)
Comment (Celsius273) [reply site/en/docs/extensions/reference/declarativeNetRequest/index.md:43]: This is more of a lower limit. so I'll use "at least" here

File: site/en/docs/extensions/reference/declarativeNetRequest/index.md (hunk 2 above)
Comment (Celsius273) [reply site/en/docs/extensions/reference/declarativeNetRequest/index.md:51]: Done

File: site/en/docs/extensions/reference/declarativeNetRequest/index.md (hunk 3 above)
Comment (Celsius273) [reply site/en/docs/extensions/reference/declarativeNetRequest/index.md:53]: Done

File: site/en/docs/extensions/reference/declarativeNetRequest/index.md (hunk 4 above)
Comment (Celsius273) [reply site/en/docs/extensions/reference/declarativeNetRequest/index.md:58]: done

File: site/en/docs/extensions/reference/declarativeNetRequest/index.md (hunk 5 above)
Comment (Celsius273) [reply site/en/docs/extensions/reference/declarativeNetRequest/index.md:59]: I thought the next few lines gives 4 kinds of rule types?

File: site/en/docs/extensions/reference/declarativeNetRequest/index.md (hunk 6 above)
Comment (Celsius273) [reply site/en/docs/extensions/reference/declarativeNetRequest/index.md:91]: thx

**Comment by Celsius273** [comment]
Done addressing comments, PTAL for another round

**Review by awfuchs — APPROVED** [review]

**Pull request merged**